src/modules/mod.rs

```diff
@@ -55,65 +55,73 @@ mod battery;
+        match module {
+            // Keep these ordered alphabetically.
+            // Default ordering is handled in configs/starship_root.rs
+            "aws" => aws::module(context),
+            #[cfg(feature = "battery")]
+            "battery" => battery::module(context),
+            "character" => character::module(context),
+            "cmake" => cmake::module(context),
+            "cmd_duration" => cmd_duration::module(context),
+            "conda" => conda::module(context),
+            "dart" => dart::module(context),
+            "directory" => directory::module(context),
+            "docker_context" => docker_context::module(context),
+            "dotnet" => dotnet::module(context),
+            "elixir" => elixir::module(context),
+            "elm" => elm::module(context),
+            "erlang" => erlang::module(context),
+            "env_var" => env_var::module(context),
+            "gcloud" => gcloud::module(context),
+            "git_branch" => git_branch::module(context),
+            "git_commit" => git_commit::module(context),
+            "git_state" => git_state::module(context),
+            "git_status" => git_status::module(context),
+            "golang" => golang::module(context),
+            "helm" => helm::module(context),
+            "hg_branch" => hg_branch::module(context),
+            "hostname" => hostname::module(context),
+            "java" => java::module(context),
+            "jobs" => jobs::module(context),
+            "julia" => julia::module(context),
+            "kubernetes" => kubernetes::module(context),
+            "line_break" => line_break::module(context),
+            "memory_usage" => memory_usage::module(context),
+            "nim" => nim::module(context),
+            "nix_shell" => nix_shell::module(context),
+            "nodejs" => nodejs::module(context),
+            "ocaml" => ocaml::module(context),
+            "package" => package::module(context),
+            "perl" => perl::module(context),
+            "php" => php::module(context),
+            "purescript" => purescript::module(context),
+            "python" => python::module(context),
+            "ruby" => ruby::module(context),
+            "rust" => rust::module(context),
+            "shlvl" => shlvl::module(context),
+            "singularity" => singularity::module(context),
+            "swift" => swift::module(context),
+            "terraform" => terraform::module(context),
+            "time" => time::module(context),
+            "crystal" => crystal::module(context),
+            "username" => username::module(context),
+            "zig" => zig::module(context),
+            _ => {
+                eprintln!("Error: Unknown module {}. Use starship module --list to list out all supported modules.", module);
+                None
+            }
         }
-    }
+    };
+
+    m.get_or_insert(context.new_module(module)).duration = Option::from(start.elapsed());
```

Comment:
```suggestion
    let elapsed = start.elapsed();
    // create empty module with timing information if it takes too long, even if it returns nothing
    if elapsed.as_millis() > 0 {
        m = m.or_else(|| context.new_module(module).duration = Option::from(elapsed));
    }
```
I think with this keeping it as `Option` makes sense.

Reply:
wrote it a bit differently because as far as I could understand your code, it only set the duration for the None case.

Reply:
One bad side effect of this: if a module is now returning None, this code makes it more or less random if this is then a `None` or a `Some(Module)`, depending if it took more or less than a 1ms. I'm not familiar enough with the tests if this might result in flaky tests :-(

Reply:
The tests ran successfully with and without that change so it seems to be fine. If it becomes an issues `Module` could be extended with an extra poperty to tell if it was created to provide timing information only.